File: 2070-most-beautiful-item-for-each-query/2070-most-beautiful-item-for-each-query.cpp
class Solution {
public:
    vector<int> maximumBeauty(vector<vector<int>>& items, vector<int>& queries) {
        int n = items.size(), m = queries.size();
        sort(items.begin(), items.end());
        vector<int> ans(m);

        vector<int> mostBeauty(n);
        mostBeauty[0] = items[0][1];
        for (int i = 1; i < n; i++) mostBeauty[i] = max(mostBeauty[i-1], items[i][1]);

        for (int i = 0; i < m; i++) {
            int l = 0, r = n;
            while (l < r) {
                int mid = l + (r - l) / 2;
                if (items[mid][0] > queries[i]) r = mid;
                else l = mid + 1;
            }
            
            if (l == 0) ans[i] = 0;
            else ans[i] = mostBeauty[l - 1];
        }

        return ans;
    }
};

File: 2070-most-beautiful-item-for-each-query/2070-most-beautiful-item-for-each-query.js
/**
 * @param {number[][]} items
 * @param {number[]} queries
 * @return {number[]}
 */
var maximumBeauty = function(items, queries) {
    var arr = [], vlarr, max;
    items.sort((a,b) => b[1] - a[1]);
    for (var i of queries) {
        vlarr = items.find(v => v[0] <=i);
        if (vlarr) {
            arr.push(vlarr[1])
        }else {
            arr.push(0)
        }
    }
    return arr.length ? arr : [0]
};